modules/subscriber/variables.tf

```diff
@@ -0,0 +1,82 @@
+variable "name" {
+  type        = string
+  description = <<-EOF
+    Name of role. Since this name must be unique within the
+    account, it will be reused for most of the resources created by this
+    module.
+  EOF
+
+  validation {
+    condition     = length(var.name) <= 64
+    error_message = "Name must be under 64 characters."
+  }
+}
+
+variable "destination_iam_arn" {
+  description = "ARN for destination iam policy"
+  type        = string
+}
+
+variable "firehose_arn" {
+  description = "ARN for kinesis firehose"
+  type        = string
+}
+
+variable "filter_name" {
+  description = "Subscription filter name. Existing filters that have this name as a pre  fix will be removed."
+  type        = string
+}
+
+variable "log_group_name_patterns" {
+  description = "List of patterns as strings. We will only subscribe to log groups that have names matching one of the provided strings based on strings based on a case-sensitive substring search. To subscribe to all log groups, use the wildcard operator *."
+  type        = list(string)
+}
+
+variable "log_group_name_prefixes" {
+  description = "List of prefixes as strings. The lambda function will only apply to log groups that start with a provided string. To subscribe to all log groups, use the wildcard operator *."
+  type        = list(string)
+
+  validation {
+    condition = alltrue([
+      for prefix in var.log_group_name_prefixes : (length(prefix) < 512 && can(regex("^(?!aws)([a-zA-Z0-9_\\-\\/\\.\\#]){1,512}", prefix))) || prefix == "*"
+    ])
+    error_message = "Invalid group name prefix provided. See https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_CreateLogGroup.html for log group name restrictions."
+  }
+
+}
+
+variable "discovery_rate" {
+  description = "EventBridge rate expression for periodically triggering discovery. If not set, no eventbridge rules are configured."
+  type        = string
+}
+
+variable "filter_pattern" {
+  description = "Subscription filter pattern."
+  type        = string
+}
+
+variable "num_workers" {
+  description = "Maximum number of concurrent workers when processing log groups."
+  type        = string
```

Comment:
Should be a number, even if it's later interpolated as a string. That'll protect you from someone doing `num_workers = "wrong"`.